parser.go

```diff
@@ -38,13 +38,35 @@ func (e *DicomElement) getTag() string {
 	return fmt.Sprintf("(%4x,%4x)", e.Group, e.Element)
 }
 
-func NewParser(dictionary io.Reader) (*Parser, error) {
-	p := &Parser{}
-	err := p.loadDictionary(dictionary)
+// Create a new parser, with functional options for configuration
+// http://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+func NewParser(options ...func(*Parser) error) (*Parser, error) {
+
+	p := Parser{}
+
+	// apply defaults
+	fh, err := os.Open("dicom.dic")
+	defer fh.Close()
+
+	if err != nil {
+		panic(err)
+	}
+
+	err = Dictionary(fh)(&p)
```

Comment:
Unfortunately, this approach reads in the default dictionary before it applies the custom dictionary (when it loops over the options functions).

Which will be a few nanoseconds slower for users who want to supply their own dictionary.

Any ideas, @Bitbored ?